### ISSUE TITLE
Supervision loss in code and paper

### ISSUE THREAD

**Issue by leilai125** [yd-yin/FisherMatch#4]
Hi, thanks for your work. It's very interesting. I realized that during supervision part, the loss function is different between the code and paper, right? In the paper, the loss function for supervision is simple (negative log likelihood of the groundtruth rotation in the predicted distributions). Why the loss function in the code is different? Does this loss function (in the code) have been mentioned in the paper?

Thanks!

**Comment by yd-yin** [comment]
Hi, thanks for your interest.

The supervised loss in the code is the negative log-likelihood, the same as the paper. It is defined at https://github.com/yd-yin/FisherMatch/blob/main/fisher/fisher_utils.py#L19

Why do you think it is different from the paper?

**Comment by leilai125** [comment]
> 

Hi I think I am a bit confused about the code is because it looks a bit different from the function (6) in the paper. What does log_exponent, overreg, log_normalizer represent respectively? I thought loss function will look like: -log(MF(y, A)) 

I also run the code using this loss function, the return loss is around negative 5. Is this normal? Loss is usually set to be positive (negative is fine as well), just want to double check

I think it might because I am not familiar with fisher matrix distribution. Hope you will provide some explanation to the code here: https://github.com/yd-yin/FisherMatch/blob/main/fisher/fisher_utils.py#L19

Thanks!

**Comment by yd-yin** [comment]
Given $p(R;A) = \frac{1}{F(A)}\exp(tr(A^TR))$, $-\log(p)=\log(F(A))-tr(A^TR)$, `log_normalizer` is $\log(F)$, `log_exponent` is $-tr(A^TR)$. `overreg` is very close to 1, which is used to better stabilize the training. This term is proposed in [1] (see last paragraph of section 3.2 in [1]) and I leave it as is.

It's normal. $-\log(p)$ will be negative if $p$ is larger than 1. (Note that $p$ is pdf, not probability)

[1] Probabilistic orientation estimation with matrix Fisher distributions. NeurIPS 2020
